### ISSUE TITLE
Invalid position given by 'no-multiple-empty-lines'

### ISSUE THREAD

**Issue by isiahmeadows** [AtomLinter/linter-eslint#1135]
ESLint returned a point that did not exist in the document being edited.
Rule: `no-multiple-empty-lines`
Requested range: 10:1

Code:

```js
"use strict"

if (require.main === module) {
    throw new Error("This isn't a runnable script!")
}

const path = require("path").posix
const cache = new Map()


function getRegExp(name) {
    if (cache.has(name)) return cache.get(name)

    const escaped = name.replace(/[|\\{}()[\]^$+*?.]/g, "\\$&")
    const re = new RegExp(`\\b${escaped}\\b`)

    cache.set(name, re)
    return re
}
```

Debug information:
```json
{
  "atomVersion": "1.26.1",
  "linterEslintVersion": "8.4.1",
  "linterEslintConfig": {
    "fixOnSave": true,
    "rulesToDisableWhileFixing": [
      "no-debugger"
    ],
    "rulesToSilenceWhileTyping": [
      "no-trailing-spaces",
      "padded-blocks"
    ],
    "lintHtmlFiles": false,
    "useGlobalEslint": false,
    "showRuleIdInMessage": true,
    "disableWhenNoEslintConfig": true,
    "eslintrcPath": "",
    "globalNodePath": "",
    "advancedLocalNodeModules": "",
    "eslintRulesDirs": [],
    "disableEslintIgnore": false,
    "disableFSCache": false,
    "scopes": [
      "source.js",
      "source.jsx",
      "source.js.jsx",
      "source.babel",
      "source.js-semantic"
    ],
    "ignoreFixableRulesWhileTyping": false
  },
  "eslintVersion": "2.13.1",
  "hoursSinceRestart": 6,
  "platform": "darwin",
  "eslintType": "local project",
  "eslintPath": "$REPO/node_modules/eslint",
  "editorScopes": [
    "source.js",
    "string.quoted.double.js",
    "punctuation.definition.string.begin.js"
  ]
}
```

**Comment by isiahmeadows** [comment]
BTW, it's still auto-fixable even in Atom. It's just not working correctly when it tries to display the lines as I type (when it's not saved).

**Comment by skylize** [comment]
Could you create a minimal repro project for this?

**Comment by Arcanemagus** [comment]
Please update the version of ESLint you are using ([v1 -> v2](https://eslint.org/docs/user-guide/migrating-to-2.0.0), [v2 -> v3](http://eslint.org/docs/user-guide/migrating-to-3.0.0), [v3 -> v4](http://eslint.org/docs/user-guide/migrating-to-4.0.0)), this particular rule was fixed in ESLint version [v3.5.0](http://eslint.org/blog/2016/09/eslint-v3.5.0-released).

Thanks for taking the time to report this though!
